Add useSettings hook for reading multiple settings at once (#55337)
WordPress/gutenberg: packages/block-editor/README.md

### useSetting

> **Deprecated**

> 

Reviewer: Should we add a note right here that it's deprecated in favor of `useSettings()`?

Author: Note added :+1:

-   `any`: Returns the value defined for the setting.

### useSettings

> Hook that retrieves the given setting for the block instance in use.
> 
> It looks up the settings first in the block instance hierarchy. If none is found, it'll look it up in the block editor store.
> 
> _Parameters_
> 
> -   _paths_ `string[]`: The path to the setting.
> 
> _Returns_
> 
> -   `any[]`: Returns the values defined for the settings.

Reviewer: ```suggestion
Hook that retrieves the given settings for the block instance in use.

It looks up the settings first in the block instance hierarchy. If none are found, it'll look them up in the block editor store.

_Parameters_

-   _paths_ `string[]`: The paths to the settings.

_Returns_

-   `any[]`: Returns the values defined for the settings.
```

-   `any`: Returns the value defined for the setting.

### useSettings

Hook that retrieves the given settings for the block instance in use.

It looks up the settings first in the block instance hierarchy. If none are found, it'll look them up in the block editor settings.

_Usage_

```js

> const [ fixed, sticky ] = useSettings( 'position.fixed', 'position.sticky' );

Reviewer 2: I'm a bit confused here because the PR description says

```js
const [ allowFixed, allowSticky ] = useSettings( [ 'position.fixed', 'position.sticky' ] );
```

Can both be used? It would be nice to be consistent so we don't need to normalise.

Author: Only the `useSettings( 'a', 'b' )` form can be used. The PR originally started with an array argument, `useSettings( settings: string[] )`, but then was changed to `useSettings( ...settings: string[] )` during review. That's why the PR description is out of date.

The main reason for that was that when reading a single setting, `useSetting( 'a' )` and `useSettings( 'a' )` are the same, except that for `useSettings` the return value is always an array (with one element).
